iroh-net/src/magicsock/node_map/node_state.rs

```diff
@@ -1185,29 +1188,58 @@ pub enum ControlMsg {
     CallMeMaybe,
 }
 
-/// Information about a direct address.
+/// Information about a *direct address*.
+///
+/// The *direct addresses* of an iroh-net node are those that could be used by other nodes to
+/// establish direct connectivity, depending on the network situation. Due to NAT configurations,
+/// for example, not all direct addresses of a node are usable by all peers.
 #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
 pub struct DirectAddrInfo {
-    /// The address reported.
+    /// The UDP address reported by the remote node.
     pub addr: SocketAddr,
-    /// The latency to the address, if any.
+    /// The latency to the remote node over this network path.
+    ///
+    /// If there has never been any connectivity via this address no latency will be known.
     pub latency: Option<Duration>,
-    /// Last control message received by this node.
+    /// Last control message received by this node about this address.
+    ///
+    /// This contains the elapsed duration since the control message was received and the
+    /// kind of control message received at that time.  Only the most recent control message
+    /// is returned.
+    ///
+    /// Note that [`ControlMsg::CallMeMaybe`] is received via a relay path, while
+    /// [`ControlMsg::Ping`] and [`ControlMsg::Pong`] are received on the path to
+    /// [`DirectAddrInfo::addr`] itself and thus convey very different information.
     pub last_control: Option<(Duration, ControlMsg)>,
-    /// How long ago was the last payload message for this node.
+    /// Elapsed time since the last payload message was received on this network path.
+    ///
+    /// This indicates how long ago a QUIC datagram was received from the remote node sent
+    /// from this [`DirectAddrInfo::addr`].  It indicates the network path was in use to
+    /// transport payload data.
     pub last_payload: Option<Duration>,
-    /// When was this connection last alive, if ever.
+    /// Elapsed time since this network path was known to exist.
+    ///
+    /// A network path be considered to exist only because the remote node advertised it.
+    /// It may not mean the path is usable.  However if there was any communication with the
+    /// remote node over this network path it also means the path exists.
+    ///
+    /// The elapsed time since *any* confirmation of the path's existence was received is
+    /// returned.  If the remote node moved network and no longer has this path, this could
+    /// be a long duration.  If the path was added via [`Endpoint::add_node_addr`] or some
+    /// node discovery the path may never have been known to exist.
```

Comment:
```suggestion
    /// returned.  If the remote node moved networks and no longer has this path, this could
    /// be a long duration.  If the path was added via [`Endpoint::add_node_addr`] or some
    /// node discovery the path may never have been known to exist.
```